components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/connector/recaptcha/ResendConfirmationReCaptchaConnector.java

```diff
@@ -0,0 +1,128 @@
+    }
+
+    @Override
+    public int getPriority() {
+
+        return 3;
+    }
+
+    @Override
+    public boolean canHandle(ServletRequest servletRequest, ServletResponse servletResponse) throws CaptchaException {
+
+        String path = ((HttpServletRequest) servletRequest).getRequestURI();
+        Property[] connectorConfigs;
+        String enable = null;
+
+        if (StringUtils.isBlank(path) || !(CaptchaUtil.isPathAvailable(path, RESEND_CONFIRMATION_URL))) {
+            return false;
+        }
+
+        try {
+            connectorConfigs = CaptchaUtil.getConnectorConfigs(servletRequest, identityGovernanceService,
+                    RESEND_CONFIRMATION_RECAPTCHA_ENABLE);
+        } catch (Exception e) {
+            // Can happen due to invalid tenant/ invalid configuration
+            if (log.isDebugEnabled()) {
+                log.debug("Unable to load connector configuration.", e);
+            }
+            return false;
+        }
+
+        for (Property connectorConfig : connectorConfigs) {
+            if ((RESEND_CONFIRMATION_RECAPTCHA_ENABLE).equals(connectorConfig.getName())) {
+                enable = connectorConfig.getValue();
+            }
+        }
+        return Boolean.parseBoolean(enable);
+    }
+
+    @Override
+    public CaptchaPreValidationResponse preValidate(ServletRequest servletRequest, ServletResponse servletResponse)
+            throws CaptchaException {
+
+        CaptchaPreValidationResponse preValidationResponse = new CaptchaPreValidationResponse();
+        String path = ((HttpServletRequest) servletRequest).getRequestURI();
+        HttpServletResponse httpServletResponse = ((HttpServletResponse) servletResponse);
+
+        if (CaptchaUtil.isPathAvailable(path, RESEND_CONFIRMATION_URL)) {
+            httpServletResponse.setHeader("reCaptchaResend", "true");
+            preValidationResponse.setCaptchaValidationRequired(true);
+            Map<String, String> params = new HashMap<>();
+            params.put("errorMessage", "recaptcha.fail.message");
+            preValidationResponse.setCaptchaAttributes(params);
+            preValidationResponse.setCaptchaValidationRequired(true);
+        }
+        return preValidationResponse;
+    }
+
+    @Override
+    public boolean verifyCaptcha(ServletRequest servletRequest, ServletResponse servletResponse)
+            throws CaptchaException {
+
+        String reCaptchaResponse = ((HttpServletRequest) servletRequest).getHeader("g-recaptcha-response");
+        if (StringUtils.isBlank(reCaptchaResponse)) {
+            throw new CaptchaClientException("reCaptcha response is not available in the request.");
+        }
+        return CaptchaUtil.isValidCaptcha(reCaptchaResponse);
+    }
+
+    @Override
+    public CaptchaPostValidationResponse postValidate(ServletRequest servletRequest, ServletResponse servletResponse)
+            throws CaptchaException {
+
+        //No need post validation.
```

Comment:
Let's add a space between slashes and the beginning of the comment.

Reply:
Fixed in #474